### DISCUSSION TITLE
Save user_id for Resource with BelongsTo

### DISCUSSION THREAD

**Discussion by Waleedviews** [filamentphp/filament#5642] [Help]
I am creating a form based on belongsTo relationship, everything is working very well except storing user_id in relation table.

```
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Section::make('User Details')
                    ->relationship('user')
                    ->columns(2)
                    ->schema([
                        TextInput::make('name'),
                        TextInput::make('email')
                            ->required()
                            ->unique(User::class),
                    ]),


                //saving above relationship id in my resource Record
                Forms\Components\Hidden::make('user_id')
                    ->dehydrateStateUsing(fn ($state) => $state->user()->id),
            ]);
    }
```

but it's not working, i am actually creating a user profile on Resource Record form, so i wanted to add id to my related table, Any help please?

thank you so much.

**Comment by danharrin** [comment]
You don't need the hidden field, `user_id` will be set but the section since it uses `relationship('user')`

**Reply by Waleedviews** [reply]
could you please elaborate it more? you mean `user_id` hidden field should be remove? and it will store the id in the `user_id` column of my related table?

**Reply by danharrin** [reply]
Yes, have you tried?

**Reply by Waleedviews** [reply]
yes and it doesn't work

`SQLSTATE[HY000]: General error: 1364 Field 'user_id' doesn't have a default value`

**Reply by danharrin** [reply]
Make sure its $fillable and your relationship is a BelongsTo

**Reply by Waleedviews** [reply]
it's fill-able that's my Record model

```
class Record extends Model
{
    use HasFactory;
    protected $guarded = [];

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
```

and that's the User Model Relations


```
public function record()
    {
        return $this->hasOne(Record::class);
    }
```

**Reply by danharrin** [reply]
And user_id is on the records table?

**Reply by Waleedviews** [reply]
of-course it's

**Reply by danharrin** [reply]
Try making the user_id field nullable

**Reply by Waleedviews** [reply]
I have tried everything and it doesn’t work, it work with the select field but with the section etc it doesn’t. Any alternative solution?

**Reply by danharrin** [reply]
Well making the column nullable will make the error go away

**Reply by Waleedviews** [reply]
But why it doesn’t store the relationship Id? It worked quite good with the Select field, I will double check and confirm the issue again. Thank you 🙏

**Reply by Waleedviews** [reply]
Update: i change my column to nullable but issue still exist, would be great if you look at this issue. thank you.

**Reply by danharrin** [reply]
Ill need a reproduction repository to look into this

**Comment by Sabreena-Hussien** [comment]
I have the same problem, I am using tenancy , any solution ?

The Error
SQLSTATE[HY000]: General error: 1364 Field 'user_id' doesn't have a default value

`Fieldset::make('user')
                    ->relationship('user')
                    ->schema([
                        TextInput::make('name')->label('اسم المستخدم')->required(),
                        TextInput::make('email')->label('البريد الالكتروني')->email()->unique(ignoreRecord: true)->required(),
                        TextInput::make('password')->label('كلمة المرور')->password()
                            ->dehydrateStateUsing(fn (string $state): string => Hash::make($state))
                            ->dehydrated(fn (?string $state): bool => filled($state))
                            ->required(fn (string $operation): bool => $operation === 'create'),
                    ])->label('بيانات الدخول'),`
